phobos_umss.sh

```diff
@@ -0,0 +1,490 @@
+
+# This script is an implementation of an iRODS-phobos connector via a Universal Mass Storage Service plugin.
+# To be placed in /var/lib/irods/msiExecCmd_bin
+
+VERSION=v1.0
+ID=$RANDOM
+STATUS="FINE"
+LEVEL=1
+DEBUG=1
+DISPLAY=0
+DISPLAY_COLOUR=1
+DISPLAY_BLACK=0
+
+LOGFILE_FANCY=/var/lib/irods/$VERSION.log
+#LOGFILE_SOBRE=/var/lib/irods/$VERSION.txt
+sudo touch $LOGFILE_FANCY
+#sudo touch $LOGFILE_SOBRE
+sudo chown "$(whoami)":"$(whoami)" $LOGFILE_FANCY
+#sudo chown "$(whoami)":"$(whoami)" $LOGFILE_SOBRE
+
+# FANCY COLORS
+Color_Off=$(tput sgr0)
+# Regular Colors
+Black=$(tput setaf 0)
+Red=$(tput setaf 1)
+Green=$(tput setaf 2)
+#Yellow=$(tput setaf 3)
+Blue=$(tput setaf 4)
+Purple=$(tput setaf 5)
+#Cyan=$(tput setaf 6)
+#White=$(tput setaf 7)
+
+# TODO Laisser à l'utilisateur la possibilité de choisir entre dirs et tapes, soit par un paramètre, soit en créant 2 tiers de stockage phobos (chacun serait en relation avec un cache, et chacun serait dans une compound resource distincte).
+
+# function for the synchronization of file $1 on local disk resource to file $2 in phobos
+syncToArch () {
+    # <your command or script to copy from cache to phobos> $1 $2
+    # e.g: /usr/local/bin/rfcp $1 rfioServerFoo:$2
+    # Syntax remainder : phobos put </path/to/local/file> <object_id>
+    
+    save_log syncToArch "BEGIN syncToArch($*)"
+    error_1=0
+    if [ "$1" ]; then
+        if [ "$2" ]; then
+            op_ph=$(type -P phobos)
+            md_script="get_metadata ${1}"
+            md=$($md_script)
+            error_1=$?
+            save_log syncToArch "$md_script return with integer ${error_1} and value \n${md}"
+            if [ $error_1 == 0 ]; then
+                echo "UNIVMSS $op_ph \"$1\" \"$2\""
+                put_script="sudo $op_ph put --metadata ${md} -f dir ${1} ${2}"
+                put=$($put_script >> $LOGFILE_FANCY 2>&1)
+                error_1=$?
+                save_log syncToArch "$put_script return with status=$put($error_1)"
+	        fi
+        else
+            save_log syncToArch "No OID given to put data \"$1\" to"
+            error_1=1
+        fi
+    else
+        save_log syncToArch "No object given to put"
+        error_1=2
+    fi
+
+    if [ "$error_1" != 0 ]; then
+        STATUS="FAILURE"
+        save_log syncToArch "STATUS=$STATUS($error_1)"
+    fi
+    save_log syncToArch "END syncToArch($*)"
+    return $error_1
+}
+
+# function for staging a file $1 from phobos to file $2 on disk
+stageToCache () {
+    # <your command to stage from phobos to cache> $1 $2
+    # Syntax Remainder : $ phobos get <object_id> </path/to/local/output/file>
+    # e.g: /usr/local/bin/rfcp rfioServerFoo:$1 $2
+    
+    save_log stageToCache "BEGIN stageToCache($*)"
+    error_2=0
+    if [ "$1" ]; then
+        if [ "$2" ]; then
+            op_ph=$(type -P phobos)
+            echo "UNIVMSS $op_ph \"$1\" \"$2\""
+            # Doit être retravaillé !
```

Comment:
minor: to remove

Reply:
Done.
The comment was no longer relevant.